### ISSUE TITLE
 3.2.0 error - could not load core assembly or one of the dependencies

### ISSUE THREAD

**Issue by mbelfer** [asudbury/NinjaCoderForMvvmCross#153]
newly generated projects using latest 3.2.0 crushing when attempting to run on IOS but work fine on Android.
The exception error is happening during call to LoadFromXaml(view) and saying "could not load core assembly or one of the dependencies for core dll" which is part of the same solution.

Here is the video of the problem:
https://dl.dropboxusercontent.com/u/96783283/ShareX/2015/01/2015-01-06_19-06-44.mp4

Adrian,
funny that you asked about the tool, it came from the following source:

Stuart Lodge ‏@slodge  Dec 29
How have I never seen ShareX before? http://bit.ly/getshrx  - fab Windows screenshot and ani-gif maker. C# source https://github.com/sharex

**Comment by asudbury** [comment]
Unfortunately my MacBookAir is not working at the moment - promise to get it working at the weekend!!!

Even then I struggle to get it working as a build host from visual studio :-(

ill be honest the last few/several internal builds I have tested against android and windows phone and not iOS! whoops!!

I find the whole iOS simulator from visual studio thing frustrating!!!

I'm thinking its something to do with the iOS project and not the core project or forms project - because as you say the android project works ok.

It could be a Xamarin issue - can you try a Xamarin XAML sample and run in your iOS simulator (must be a XAML sample).

Ideally one that does data binding to a view model.

thanks

adrian.

**Comment by mbelfer** [comment]
adrian,

what would you like me to try? Some Xamarin.Forms xaml program that was created without ninja tool?
if that is the case - all works good. I been using previous versions of Ninja tool for a while now and never had any problems as well as with Xamarin.Forms using xaml that were created with VS.NET 2013

**Comment by lothrop** [comment]
I can reproduce and can confirm that http://codeworks.it/blog/?p=242 fixes it. Any ideas for a more elegant solution?

**Comment by asudbury** [comment]
brilliant work Kerry (again).

looks like I can fix with an update to the nuget package (i hope!) rather than an update to the ninja installer.

will take a look tomorrow.

thanks again.

**Comment by mbelfer** [comment]
in my case xamarin.Behaviors was not part of the error, so it could be some other xamarin dll.
anyway, i tried the workaround and the problem is still there (unless I am interpreting workaround wrong)

**Comment by lothrop** [comment]
Sorry, I didn't mean the exact workaround for Xamarin.Forms.Behaviors. I meant referencing something from the Core assembly in the iOS assembly, e.g., creating an instance of a ViewModel in `Main()`.

**Comment by asudbury** [comment]
I could just create an an Init method in the core module and the do core.Init() from the iOS FinishedLaunching method - would this work?

**Comment by mbelfer** [comment]
that worked for me ;~)

**Comment by lothrop** [comment]
Sounds good. I wish it weren't necessary.

**Comment by asudbury** [comment]
I have updated the relevant nuget packages - could someone do a quick test for me (as my MBA is broke!) - then I can close this issue and move on to v3.3.0.

thanks for your help @lothrop and @mbelfer - this shouldn't have happened :-(

**Comment by lothrop** [comment]
I tried testing with Xamarin.Forms and Unified API/prelease packages. I think you should be able to reproduce these errors even without a Mac:

Install-Package Scorchio.NinjaCoder.Xamarin.Forms.Core -FileConflictAction ignore -ProjectName FormsTest4.Core 
Install-Package Moq -FileConflictAction ignore -ProjectName FormsTest4.Core.Tests 
Install-Package Scorchio.NinjaCoder.NUnit.Tests -FileConflictAction ignore -ProjectName FormsTest4.Core.Tests 
Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -ProjectName FormsTest4.Forms 
Install-Package Scorchio.NinjaCoder.Xamarin.Forms -FileConflictAction ignore -ProjectName FormsTest4.Forms 
Install-Package Moq -FileConflictAction ignore -ProjectName FormsTest4.Forms.Tests 
Install-Package Scorchio.NinjaCoder.NUnit.Tests -FileConflictAction ignore -ProjectName FormsTest4.Forms.Tests 
Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -ProjectName FormsTest4.Droid 
Install-Package Scorchio.NinjaCoder.Xamarin.Forms -FileConflictAction ignore -ProjectName FormsTest4.Droid 
Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -ProjectName FormsTest4.iOS 
Install-Package Scorchio.NinjaCoder.Xamarin.Forms.iOS.Unified -FileConflictAction ignore -ProjectName FormsTest4.iOS 
Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -ProjectName FormsTest4.WindowsPhone 
Install-Package Scorchio.NinjaCoder.Xamarin.Forms -FileConflictAction ignore -ProjectName FormsTest4.WindowsPhone 

$DTE.ItemOperations.OpenFile('C:\test\FormsTest4\NinjaReadMe.txt') 
Installing 'Scorchio.NinjaCoder.Xamarin.Forms.Core 1.0.5'.
Successfully installed 'Scorchio.NinjaCoder.Xamarin.Forms.Core 1.0.5'.
Adding 'Scorchio.NinjaCoder.Xamarin.Forms.Core 1.0.5' to FormsTest4.Core.
Successfully added 'Scorchio.NinjaCoder.Xamarin.Forms.Core 1.0.5' to FormsTest4.Core.

Installing 'Moq 4.2.1409.1722'.
Successfully installed 'Moq 4.2.1409.1722'.
Adding 'Moq 4.2.1409.1722' to FormsTest4.Core.Tests.
Successfully added 'Moq 4.2.1409.1722' to FormsTest4.Core.Tests.

Attempting to resolve dependency 'NUnit'.
Installing 'NUnit 2.6.4'.
Successfully installed 'NUnit 2.6.4'.
Installing 'Scorchio.NinjaCoder.NUnit.Tests 1.0.2'.
Successfully installed 'Scorchio.NinjaCoder.NUnit.Tests 1.0.2'.
Adding 'NUnit 2.6.4' to FormsTest4.Core.Tests.
Successfully added 'NUnit 2.6.4' to FormsTest4.Core.Tests.
Adding 'Scorchio.NinjaCoder.NUnit.Tests 1.0.2' to FormsTest4.Core.Tests.
Successfully added 'Scorchio.NinjaCoder.NUnit.Tests 1.0.2' to FormsTest4.Core.Tests.

Installing 'Xamarin.Forms 1.3.1.6294-pre1'.
Successfully installed 'Xamarin.Forms 1.3.1.6294-pre1'.
Adding 'Xamarin.Forms 1.3.1.6294-pre1' to FormsTest4.Forms.
Install failed. Rolling back...
Install-Package : A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number, in condition 
"'$(TargetPlatformVersion)' > '8.0'".  C:\Program Files (x86)\MSBuild\12.0\bin\Microsoft.Common.CurrentVersion.targets
At line:4 char:1
+ Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -Pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidProjectFileException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
Attempting to resolve dependency 'Xamarin.Forms'.
Installing 'Scorchio.NinjaCoder.Xamarin.Forms 1.0.7'.
Successfully installed 'Scorchio.NinjaCoder.Xamarin.Forms 1.0.7'.
Adding 'Scorchio.NinjaCoder.Xamarin.Forms 1.0.7' to FormsTest4.Forms.
Successfully added 'Scorchio.NinjaCoder.Xamarin.Forms 1.0.7' to FormsTest4.Forms.

'Moq 4.2.1409.1722' already installed.
Adding 'Moq 4.2.1409.1722' to FormsTest4.Forms.Tests.
Install-Package : Failed to add reference to 'Moq'.
At line:6 char:1
+ Install-Package Moq -FileConflictAction ignore -ProjectName FormsTest4.Forms.Tes ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
Attempting to resolve dependency 'NUnit'.
'Scorchio.NinjaCoder.NUnit.Tests 1.0.2' already installed.
Adding 'NUnit 2.6.4' to FormsTest4.Forms.Tests.
Install-Package : Failed to add reference to 'nunit.framework'.
At line:7 char:1
+ Install-Package Scorchio.NinjaCoder.NUnit.Tests -FileConflictAction ignore -Proj ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
Attempting to resolve dependency 'Xamarin.Android.Support.v4 (≥ 19.1.0)'.
Installing 'Xamarin.Android.Support.v4 19.1.0'.
Successfully installed 'Xamarin.Android.Support.v4 19.1.0'.
Adding 'Xamarin.Android.Support.v4 19.1.0' to FormsTest4.Droid.
Uninstalling 'Xamarin.Android.Support.v4 19.1.0'.
Successfully uninstalled 'Xamarin.Android.Support.v4 19.1.0'.
Install failed. Rolling back...
Install-Package : Failed to add reference to 'Xamarin.Android.Support.v4'.
At line:8 char:1
+ Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -Pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
Attempting to resolve dependency 'Xamarin.Forms'.
Attempting to resolve dependency 'Xamarin.Android.Support.v4 (≥ 19.1.0)'.
Installing 'Xamarin.Android.Support.v4 19.1.0'.
Successfully installed 'Xamarin.Android.Support.v4 19.1.0'.
Adding 'Xamarin.Android.Support.v4 19.1.0' to FormsTest4.Droid.
Uninstalling 'Xamarin.Android.Support.v4 19.1.0'.
Successfully uninstalled 'Xamarin.Android.Support.v4 19.1.0'.
Install failed. Rolling back...
Install-Package : Failed to add reference to 'Xamarin.Android.Support.v4'.
At line:9 char:1
+ Install-Package Scorchio.NinjaCoder.Xamarin.Forms -FileConflictAction ignore -Pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
'Xamarin.Forms 1.3.1.6294-pre1' already installed.
Adding 'Xamarin.Forms 1.3.1.6294-pre1' to FormsTest4.iOS.
Install-Package : A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "" instead of a number, in condition 
"'$(TargetPlatformVersion)' > '8.0'".  C:\Program Files (x86)\MSBuild\12.0\bin\Microsoft.Common.CurrentVersion.targets
At line:10 char:1
+ Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -Pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidProjectFileException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
Installing 'Scorchio.NinjaCoder.Xamarin.Forms.iOS.Unified 1.0.4'.
Successfully installed 'Scorchio.NinjaCoder.Xamarin.Forms.iOS.Unified 1.0.4'.
Adding 'Scorchio.NinjaCoder.Xamarin.Forms.iOS.Unified 1.0.4' to FormsTest4.iOS.
Successfully added 'Scorchio.NinjaCoder.Xamarin.Forms.iOS.Unified 1.0.4' to FormsTest4.iOS.

Attempting to resolve dependency 'WPtoolkit (≥ 4.2013.08.16)'.
Installing 'WPtoolkit 4.2013.08.16'.
Successfully installed 'WPtoolkit 4.2013.08.16'.
Adding 'WPtoolkit 4.2013.08.16' to FormsTest4.WindowsPhone.
Uninstalling 'WPtoolkit 4.2013.08.16'.
Successfully uninstalled 'WPtoolkit 4.2013.08.16'.
Install failed. Rolling back...
Install-Package : Failed to add reference to 'Microsoft.Phone.Controls.Toolkit'.
At line:12 char:1
+ Install-Package Xamarin.Forms –IncludePrerelease  -FileConflictAction ignore -Pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 
Attempting to resolve dependency 'Xamarin.Forms'.
Attempting to resolve dependency 'WPtoolkit (≥ 4.2013.08.16)'.
Installing 'WPtoolkit 4.2013.08.16'.
Successfully installed 'WPtoolkit 4.2013.08.16'.
Adding 'WPtoolkit 4.2013.08.16' to FormsTest4.WindowsPhone.
Uninstalling 'WPtoolkit 4.2013.08.16'.
Successfully uninstalled 'WPtoolkit 4.2013.08.16'.
Install failed. Rolling back...
Install-Package : Failed to add reference to 'Microsoft.Phone.Controls.Toolkit'.
At line:13 char:1
+ Install-Package Scorchio.NinjaCoder.Xamarin.Forms -FileConflictAction ignore -Pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], InvalidOperationException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PowerShell.Commands.InstallPackageCommand
 


AutoHides             : False
Caption               : NinjaReadMe.txt
Collection            : {Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowBase, 
                        Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowBase, 
                        Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowBase, 
                        Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowBase...}
CommandBars           : 
ContextAttributes     : System.__ComObject
DTE                   : System.__ComObject
Document              : System.__ComObject
HWnd                  : 0
Height                : 404
IsFloating            : False
Kind                  : Document
Left                  : 32
Linkable              : False
LinkedWindowFrame     : Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowBase
LinkedWindows         : 
Object                : System.__ComObject
ObjectKind            : {8E7B96A8-E33D-11D0-A6D5-00C04FB67F6A}
Project               : System.__ComObject
ProjectItem           : System.__ComObject
Selection             : System.__ComObject
Top                   : 154
Type                  : vsWindowTypeDocument
Visible               : True
Width                 : 1047
WindowState           : vsWindowStateMaximize
HasBeenDeleted        : False
Events                : Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowEvents
VisibilityEvents      : Microsoft.VisualStudio.Platform.WindowManagement.DTE.WindowVisibilityEvents
Rect                  : 32,154,1047,404
OutstandingEventCount : 0

**Comment by asudbury** [comment]
I don’t think that is a ninja issue.

 

I have only updated 3 nuget packages.

 

I get similar errors from time to time. Try restating visual studio.

 

Adrian.

**Comment by mbelfer** [comment]
i just tried and newly generated project works fine for me but only after VS.NET restart just as you suggested. Had the same 3 nuget issues on a first run.
So it seems like it is fixed now.

**Comment by mbelfer** [comment]
hmmm....

ok, here is what i see

it works fine fresh after restarting vs.net but the second time it would not and the only way to get it working again is to restart vs.net

**Comment by lothrop** [comment]
Strange behavior. I'm having it, too.

I noticed one more thing: The iOS version needs to be set to at least 5.1.1 for Unified API projects. I've updated the Info.plist and issued a change request.

**Comment by asudbury** [comment]
I believe this is a known nuget issue

https://nuget.codeplex.com/workitem/3996

**Comment by asudbury** [comment]
might be worth moving the nuget (and resharper?) issue you are having to a new github issue - and close this one as the subject matter I believe is now fixed.

**Comment by asudbury** [comment]
This is item has now been fixed.
